### PR TITLE
chore: implement golang generation options and JSON serialization runtime

### PR DESCRIPTION
**Pull request by kzantow**
This PR cleans up the code a bit and the generated code is _close_ to proper go format.

### CODE REVIEW

File: src/shacl2code/lang/golang.py [hunk 1]
```diff
@@ -2,153 +2,300 @@
...
     "http://www.w3.org/2001/XMLSchema#boolean": "bool",
     "http://www.w3.org/2001/XMLSchema#decimal": "float64",
-    "http://www.w3.org/2001/XMLSchema#dateTime": "DateTime",
-    "http://www.w3.org/2001/XMLSchema#dateTimeStamp": "DateTimeStamp",
+    "http://www.w3.org/2001/XMLSchema#dateTime": "string", # "DateTime",
+    "http://www.w3.org/2001/XMLSchema#dateTimeStamp": "string", # "DateTimeStamp",
 }
 
+RESERVED_WORDS = {
+    "package"
```
Comment (nishakm) [inline src/shacl2code/lang/golang.py:26]: Why is "package" dealt with separately?

File: src/shacl2code/lang/golang.py [hunk 2]
```diff
@@ -2,153 +2,300 @@
...
+    if isinstance(name, list):
+        name = "".join(name)
+    parts = re.split(r'[^a-zA-Z0-9]', name)
+    part = parts[len(parts)-1]
+    return upper_first(part)
 
 
+def gofmt(code):
+    try:
+        proc = subprocess.Popen(["gofmt"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE, text=True)
```
Comment (nishakm) [inline src/shacl2code/lang/golang.py:295]: You could just use `subprocess.run(["gofmt"], capture_output=true)`. It's thread safe.

**Review by nishakm — COMMENTED** [review]
Can you explain what is in `exported_flat`?

File: src/shacl2code/lang/golang.py (hunk 1 above)
Comment (kzantow) [reply src/shacl2code/lang/golang.py:26]: it's a reserved word in `go`, no matter what, we can't output a lowercase name of `package`

**Comment by kzantow** [comment]
As discussed, I opened a new PR for this since I will have to iterate on it, it makes more sense not to have a requirement for you to continually merge my changes: https://github.com/JPEWdev/shacl2code/pull/21

**Pull request closed without merging**